docs/source/serving/auth.rst

```diff
@@ -0,0 +1,88 @@
+
+  run: |
+    export PATH=$PATH:/sbin
+    python -m vllm.entrypoints.openai.api_server \
+      --model $MODEL_NAME --trust-remote-code \
+      --gpu-memory-utilization 0.95 \
+      --host 0.0.0.0 --port 8000 \
+      --api-key $AUTH_TOKEN
+
+To enable this feature in SkyServe, you need to configure the readiness probe to include the access token (see the :code:`service.readiness_probe` section above). This ensures the readiness probe passes the authorization check. Use the following command to deploy the service:
+
+.. code-block:: bash
+
+  sky serve up auth.yaml -n auth --env HF_TOKEN=<your-hf-token> --env AUTH_TOKEN=sky-authkey-3d2105b9-a9ba-4f13
+
+SkyServe's proxy design ensures that all headers in the original request, including the authorization token, are forwarded to the vLLM inference engine. The engine then validates the token. Notice that we will automatically replace the :code:`$AUTH_TOKEN` in the service section with the actual token value as well.
+
+With the :code:`--api-key` set, a user's request needs to include an Authorization header to access the service:
+
+.. code-block:: bash
+
+  $ ENDPOINT=$(sky serve status --endpoint auth)
+  $ AUTH_TOKEN=sky-authkey-3d2105b9-a9ba-4f13
+  $ curl http://$ENDPOINT/v1/chat/completions \
+      -H "Content-Type: application/json" \
+      -H "Authorization: Bearer $AUTH_TOKEN" \
+      -d '{
+        "model": "meta-llama/Meta-Llama-3-8B-Instruct",
+        "messages": [
+          {
+            "role": "system",
+            "content": "You are a helpful assistant."
+          },
+          {
+            "role": "user",
+            "content": "Who are you?"
+          }
+        ],
+        "stop_token_ids": [128009, 128001]
+      }'
+  {"id":"cmpl-0e93e6ea6e9c40f5805135afec75f163","object":"chat.completion","created":1716714050,"model":"meta-llama/Meta-Llama-3-8B-Instruct","choices":[{"index":0,"message":{"role":"assistant","content":"I'm your friendly AI assistant! I'm a computer program designed to help you with a wide range of tasks and answer your questions to the best of my ability. I'm here to provide information, offer suggestions, and assist you in any way I can. I'm constantly learning and improving, so the more you interact with me, the better I'll become at understanding your needs and providing helpful responses.\n\nI can help with things like:\n\n* Answering general knowledge questions\n* Providing definitions for words and phrases\n* Giving advice on topics like science, history, and technology\n* Assisting with language-related tasks, such as grammar and proofreading\n* Generating text and writing articles or stories\n* Even having fun conversations and playing games with you!\n\nSo, what can I help you with today?"},"logprobs":null,"finish_reason":"stop","stop_reason":128009}],"usage":{"prompt_tokens":26,"total_tokens":188,"completion_tokens":162}}
+
+If the Authorization header is missing or invalid, the service will return a 401 Unauthorized error:
```

Comment:
A service client without an API key will not be able to access the service and get a :code:`401 Unauthorized` error: